src/components/Table/EditColumnsButton.tsx

```diff
@@ -11,21 +11,25 @@ import { GridColumn, Kinded } from "./GridTable";
 interface EditColumnsButtonProps<R extends Kinded, S>
   extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
   columns: GridColumn<R, S>[];
-  setColumns: Dispatch<SetStateAction<GridColumn<R, S>[]>>;
+  setColumns: (value: GridColumn<R, S>[]) => void;
   title?: string;
+  storageKey?: string;
   // for storybook purposes
   defaultOpen?: boolean;
 }
 
 export function EditColumnsButton<R extends Kinded, S = {}>(props: EditColumnsButtonProps<R, S>) {
-  const { defaultOpen, disabled, columns, setColumns, trigger, title } = props;
+  const { defaultOpen, disabled, columns, setColumns, trigger, title, storageKey } = props;
   const state = useMenuTriggerState({ isOpen: defaultOpen });
   const buttonRef = useRef<HTMLButtonElement>(null);
   const { menuTriggerProps } = useMenuTrigger({ isDisabled: !!disabled }, state, buttonRef);
   const tid = useTestIds(
     props,
     isTextButton(trigger) ? trigger.label : isIconButton(trigger) ? trigger.icon : trigger.name,
   );
+  
+  const storageData = storageKey ? (JSON.parse(sessionStorage.getItem(storageKey)!) as GridColumn<R, S>[]) : null;
+  const storageColumns = storageData && (storageData.map((column) => column.name) as string[]);
```

Comment:
**question:** Do we need to check the session storage here? I believe if we are using the columns returned by the `useColumns` hook, then we shouldn't need to look in the session storage again.